src/common/cluster-store/cluster-store.ts

```diff
@@ -111,12 +115,24 @@ export class ClusterStore extends BaseStore<ClusterStoreModel> {
     return undefined;
   }
 
+  private createNewCluster(model: ClusterModel): Cluster {
```

Comment:
Not welcome here at all. This logic belongs to `createCluster` dependency function.